test/03_USM.test.js

```diff
@@ -29,127 +43,280 @@ contract('USM', (accounts) => {
+        ethPool3.toString().should.equal(oneEth.mul(TWO).toString())
+
+        const fumBalance3 = (await fum.balanceOf(user2))
+        // Calls to fund() after the first start using sliding prices.  In particular, following the math in USM.fumFromFund(),
+        // with the second call here 2xing the pool (1 ETH -> 2) and the initial FUM price of $1 = 0.004 ETH, the FUM added should
+        // be 1 (current ETH pool) / 0.004 * (1 - 1 / 2) = 125, ie, the user's FUM balance should increase by a factor of 3/2:
+        fumBalance3.toString().should.equal(wadMul(oneEth, priceWAD).mul(THREE).div(TWO).toString())
+
+        const fundDefundAdj3 = (await usm.fundDefundAdjustment())
+        // We've 2x'd the ETH pool, which according to the adjustment math in USM.fund(), should 4x the fundDefundAdjustment:
+        const targetFundDefundAdj3 = WAD.mul(FOUR)
+        fundDefundAdj3.toString().should.equal(targetFundDefundAdj3.toString())
+
+        const fumBuyPrice3 = (await usm.fumPrice(sides.BUY))
+        const fumSellPrice3 = (await usm.fumPrice(sides.SELL))
+        // The FUM buy & sell prices will have changed now, because the sliding-price above will have effectively collected fees
+        // in the contract, which go to FUM holders.  Calculating the increase would be a bit of work, but one simple check we can
+        // apply is, fundDefundAdjustment = 4 should mean the FUM buy price is now 4x the FUM sell price:
+        fumBuyPrice3.toString().should.equal(wadMul(fumSellPrice3, fundDefundAdj3).toString())
+
+        // Check that fundDefundAdjustment decays properly (well, approximately) over time:
+        // - Our adjustment was previously 4: see targetFundDefundAdj3 above.
+        // - Use timeMachine to move forward 180 secs = 3 min.
+        // - Since USM.BUY_SELL_ADJUSTMENTS_HALF_LIFE = 60 (1 min), 3 min should mean a decay of ~1/8.
+        // - The way our "time decay towards 1" approximation works, decaying 4 by ~1/8 should yield 1 + (4 * 1/8) - 1/8 = 1.375.
+        //   (See comment in USM.mintBurnAdjustment().)
+        // - And finally - since our decay's exponentiation is approximate (WadMath.wadHalfExp()), don't compare fundDefundAdj4 to
+        //   targetFundDefundAdj4 *exactly*, but instead first round both to the nearest multiple of 10 (via divRound(10)).
+        const block0 = (await web3.eth.getBlockNumber())
+        const t0 = (await web3.eth.getBlock(block0)).timestamp
+        const timeDelay = 3 * 60
+        await timeMachine.advanceTimeAndBlock(timeDelay)
+        const block1 = (await web3.eth.getBlockNumber())
+        const t1 = (await web3.eth.getBlock(block1)).timestamp
+        t1.toString().should.equal((t0 + timeDelay).toString())
+        const fundDefundAdj4 = (await usm.fundDefundAdjustment())
+        const decayFactor4 = WAD.div(EIGHT)
+        const targetFundDefundAdj4 = WAD.add(wadMul(fundDefundAdj3, decayFactor4)).sub(decayFactor4)
```

Comment:
I think it might be useful to code this in a helper function, same as you did with `wadMul`.

Reply:
Done.